src/ert/scheduler/scheduler.py

```diff
@@ -214,35 +214,36 @@ async def gather_realization_jobs() -> list[BaseException | None]:
                 for scheduling_task in scheduling_tasks:
                     scheduling_task.cancel()
 
-        job_results: Optional[list[BaseException | None]] = None
         try:
             # there are two types of tasks and each type is handled differently:
             # -`gather_realization_jobs` does not raise; rather, each job task's result
             # is collected into the returning list. Exceptions from the job tasks are
             # handled in the `else` branch after the evaluation has stopped.
             # -If exception occurs, it must necessarily come from `scheduling tasks`
-            job_results = (
-                await asyncio.gather(gather_realization_jobs(), *scheduling_tasks)
-            )[0]
+
+            await asyncio.gather(gather_realization_jobs(), *scheduling_tasks)
+
         except (asyncio.CancelledError, Exception) as e:
-            for job_task in self._tasks.values():
-                job_task.cancel()
-                # suppress potential error during driver.kill
-                with suppress(Exception):
-                    await job_task
-            for scheduling_task in scheduling_tasks:
-                scheduling_task.cancel()
-            # Log and re-raise non-cancellation errors.
-            if not isinstance(e, asyncio.CancelledError):
-                logger.error(str(e))
-                raise e
-        else:
-            for result in job_results or []:
-                if not isinstance(result, asyncio.CancelledError) and isinstance(
-                    result, Exception
-                ):
-                    logger.error(str(result))
-                    raise result
+            if isinstance(e, asyncio.CancelledError):
+
+                for result in self._tasks.values() or []:
+                    if not isinstance(result, asyncio.CancelledError) and isinstance(
+                        result, Exception
+                    ):
+                        logger.error(str(result))
+                        raise result
+            else:
+                for job_task in self._tasks.values():
+                    job_task.cancel()
+                    # suppress potential error during driver.kill
+                    with suppress(Exception):
+                        await job_task
+                for scheduling_task in scheduling_tasks:
```

Comment:
Do we need to cancel the scheduling_tasks again? I assume it is already done on line 215.

Reply:
I think so, yes. When exception happens if one of the `scheduling_tasks` then the rest will still run and thus we need to cancel.